bundle/phases/deploy.go

```diff
@@ -19,55 +19,90 @@ import (
+func parseTerraformActions(changes []*tfjson.ResourceChange, toInclude func(typ string, actions tfjson.Actions) bool) []terraformlib.Action {
+	res := make([]terraformlib.Action, 0)
+	for _, rc := range changes {
+		if !toInclude(rc.Type, rc.Change.Actions) {
 			continue
 		}
 
 		var actionType terraformlib.ActionType
-
 		switch {
 		case rc.Change.Actions.Delete():
 			actionType = terraformlib.ActionTypeDelete
 		case rc.Change.Actions.Replace():
 			actionType = terraformlib.ActionTypeRecreate
 		default:
-			// We don't need a prompt for non-destructive actions like creating
-			// or updating a schema.
+			// No use case for other action types yet.
 			continue
 		}
 
-		actions = append(actions, terraformlib.Action{
+		res = append(res, terraformlib.Action{
 			Action:       actionType,
 			ResourceType: rc.Type,
 			ResourceName: rc.Name,
 		})
 	}
 
-	// No restricted actions planned. No need for approval.
-	if len(actions) == 0 {
+	return res
+}
+
+func approvalForDeploy(ctx context.Context, b *bundle.Bundle) (bool, error) {
+	tf := b.Terraform
+	if tf == nil {
+		return false, fmt.Errorf("terraform not initialized")
+	}
+
+	// read plan file
+	plan, err := tf.ShowPlanFile(ctx, b.Plan.Path)
+	if err != nil {
+		return false, err
+	}
+
+	schemaActions := parseTerraformActions(plan.ResourceChanges, func(typ string, actions tfjson.Actions) bool {
+		// Filter in only UC schema resources.
+		if typ != "databricks_schema" {
+			return false
+		}
+
+		// We only display prompts for destructive actions like deleting or
+		// recreating a schema.
+		return actions.Delete() || actions.Replace()
+	})
+
+	dltActions := parseTerraformActions(plan.ResourceChanges, func(typ string, actions tfjson.Actions) bool {
+		// Filter in only DLT pipeline resources.
+		if typ != "databricks_pipeline" {
+			return false
+		}
+
+		// Recreating DLT pipeline leads to metadata loss and for a transient period
+		// the underling tables will be unavailable.
+		return actions.Replace()
+	})
+
+	// We don't need to display any prompts in this case.
+	if len(dltActions) == 0 && len(schemaActions) == 0 {
 		return true, nil
 	}
 
-	cmdio.LogString(ctx, "The following UC schemas will be deleted or recreated. Any underlying data may be lost:")
-	for _, action := range actions {
-		cmdio.Log(ctx, action)
+	// One or more UC schema resources will be deleted or recreated.
+	if len(schemaActions) != 0 {
+		cmdio.LogString(ctx, "The following UC schemas will be deleted or recreated. Any underlying data may be lost:")
+		for _, action := range schemaActions {
+			cmdio.Log(ctx, action)
+		}
+	}
+
+	// One or more DLT pipelines is being recreated.
+	if len(dltActions) != 0 {
+		cmdio.LogString(ctx, "The following DLT pipelines will be recreated. Underlying tables will be unavailable for a transient period until the newly recreated pipelines are run once successfully. History of previous pipeline update runs will be lost because of recreation:")
```

Comment:
Two adjustments I'd like to see:
* This message should indicate _why_ the pipeline might be getting recreated, helping customers understand what they can do about that.
* This message should hint about the `--auto-approve` flag since users may find out about it in CI. It's pretty tricky to make approval work well in a CI environment where you might not have direct prod access, and some customers will just want to use the flag.

Reply:
1. Done
2. It's already the case (see right below). We return an error suggesting `--auto-approve` if we can't prompt (ie it's not a TTY) which is the case for CI environments.
```
the deployment requires destructive actions, but current console does not support prompting. Please specify --auto-approve if you would like to skip prompts and proceed
```